include/trieste/gen.h

```diff
@@ -0,0 +1,20 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include "ast.h"
+#include "xoroshiro.h"
+
+#include <regex>
+
+namespace trieste
+{
+  using sv_match = std::match_results<std::string_view::const_iterator>;
+
+  using Rand = xoroshiro::p128r32;
+  using Seed = uint64_t;
+  using Result = uint32_t;
+
+  using GenLocationF = std::function<std::string(Rand& rnd)>;
+  using GenNodeLocationF = std::function<Location(Rand&, Node)>;
+}
```

Comment:
Is it worth moving the Gen class from `wf.h` here as well?

Reply:
No, it's specific to wf.h. This is just the stuff that's shared between wf.h and parse.h.